x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycle.java

```diff
@@ -268,6 +270,13 @@ protected Clock getClock() {
         return actions;
     }
 
+    @Override
+    public void onIndexModule(IndexModule indexModule) {
+        if (ilmEnabled) {
+            indexModule.addIndexEventListener(indexLifecycleInitialisationService.get());
```

Comment:
Can you add an `assert indexLifecycleInitialisationService.get() != null` before this line? I want to make sure we don't add any NPEs in the future